logic/logic.py

```diff
@@ -454,7 +483,8 @@ def get_items_by_usefulness_fraction(self, item_names_to_check):
     accessible_undone_locations = self.get_accessible_remaining_locations(for_progression=True)
     inaccessible_undone_item_locations = []
     locations_to_check = self.remaining_item_locations
-    locations_to_check = self.filter_locations_for_progression(locations_to_check)
+    need_sunken_treasure = any(item.startswith("Treasure Chart ") or item.startswith("Triforce Chart ") for item in item_names_to_check)
+    locations_to_check = self.filter_locations_for_progression(locations_to_check, filter_sunken_treasure=(not need_sunken_treasure))
```

Comment:
This function is only called here outside of the starting item randomizer: https://github.com/LagoLunatic/wwrando/blob/7375ab52e4b3fd9bcf180081fa930ce5f98dcaf5/randomizers/items.py#L282-L287

I'm not completely sure of all the implications of this change but I haven't seen anything obvious in testing so far. I think the only times this would make a difference to that callsite in the items randomizer is:
* If no charts are in logic, then chart locations aren't considered for progression, so the grappling hook doesn't always get a max fraction of 2 regardless of settings (I think this is a bugfix)
* If some charts are in logic but all charts have already been placed, and the grappling hook hasn't yet, then it artificially has a lower usefulness fraction (but since it is required for ganondorf it's still <9999)

A maybe simpler/safer solution (that I'm not able to write/test properly right now, it might break other things) could be to instead in `filter_locations_for_progression_static` actually follow the treasure/triforce charts progression options and always exclude the sunken locations when the locations aren't enabled, in addition to filtering them with filter_sunken_treasure for the callers that need it

Reply:
I think the safest solution would be just adding a new `filter_sunken_treasure` parameter to `get_items_by_usefulness_fraction` that it passes down to `filter_locations_for_progression`. Then the item randomizer can pass False guaranteeing that nothing changes about the existing behavior there for now, and the starting items randomizer can pass whatever it needs to.

I'm curious about this apparently being a bugfix for the grappling hook in the item randomizer but that should be a separate PR later, I don't know what all the implications here are either since I haven't thought about this part of the code in 5 years, and I wasn't able to find a seed that seems affected by the bug you describe. But the item randomizer tends to be pretty fragile in edge cases.

Reply:
> I'm curious about this apparently being a bugfix for the grappling hook in the item randomizer

I don't think it really can realistically affect the item randomizer (and I also haven't managed to find a seed that exposes it, it would be extremely rare), I just meant a bug in that the fraction for the grappling hook returned by `get_items_by_usefulness_fraction` might take into account charts that aren't actually in logic and be artificially low

I'll test a bit more, see if I can wrangle a seed that reproduces it and send the PR if I find one